### ISSUE TITLE
Proposal: add flags to uv_run, a la ev_run

### ISSUE THREAD

**Issue by saghul** [joyent/libuv#339]
I noticed issue #291, and I think (please, correct me if I'm wrong), that the implementation of NOWAIT in Windows would just involve passing false to the call to poll here: https://github.com/joyent/libuv/blob/master/src/win/core.c#L237 (perhaps with a new parameter to UV_LOOP_ONCE, but I didn't hink too much of that yet).

So assuming that would be added there would be uv_run, uv_run_one and uv_run_nowait (or uv_run_once_nowait). What about having something like this:

```
typedef enum {
    UV_RUN_DEFAULT = 0,
    UV_RUN_ONCE = 1,
    UV_RUN_NOWAIT = 2,
} uv_run_flags;
```

And then just a single run function:

```
int uv_run(uv_loop_t *loop, int flags)
```

Thoughts? If I get an ok I could work on a patch and submit a pull request :-)

**Comment by saghul** [comment]
Changing uv_run might be too much, so maybe having 2 functions would not hurt. What about this?

```
int uv_run(uv_loop_t *loop)
int uv_run2(uv_loop_t *loop, int flags)
```

uv_run would just call uv_run2(loop, UV_RUN_DEFAULT).

If it's not a good time for this (i see 0.8 is approaching) please let me know and I'll leave this open and tackle it later :-)

Thanks,

**Comment by bnoordhuis** [comment]
I won't take patches right now but maybe I'll steal your idea and apply it in the upcoming refactor.

**Comment by saghul** [comment]
Cool!

/Saúl

On 05/04/2012, at 18:49, Ben Noordhuis
reply@reply.github.com
wrote:

**Comment by saghul** [comment]
Now that 0.8 is out can we reconsider this? /cc @bnoordhuis @piscisaureus

**Comment by bnoordhuis** [comment]
Yes. I'll add it.

**Comment by piscisaureus** [comment]
Fixed in 0820be7
